### ISSUE TITLE
telegraf agent azure output plugin does not support sovereign clouds (e.g. usgovcloud)

### ISSUE THREAD

**Issue by ernstae** [influxdata/telegraf#5452]
## Feature Request

I've been using telegraf agent to ship metrics to Azure Monitor instances in multiple Azure subscriptions.  It looks like the URL is hard-coded as [regionname].monitoring.azure.com which doesn't result in support of sovereign cloud environments like US Goverment (e.g. `usgovvirginia.monitoring.azure.com` is incorrect in this context, and should be `monitoring.core.usgovcloudapi.net`.)

### Proposal:
Allow for endpoint overrides to support the environment, with an output configuration variable to define the endpoint.

It would be even cleaner if telegraf were able to automatically determine where it was running by querying the Azure Instance Metadata service, but reviewing the available information doesn't show tenantId (_which can be used to derive whether it's Azure Commercial or Government_), or any other information to reveal the necessary information.

### Current behavior:
telegraf attempts to send metrics to `https://<region>.monitoring.azure.com`

### Desired behavior:
Allow for an override for the Azure Monitor output host.
For example, the regional endpoints seem to be shunned in favor of the simpler `monitoring.core.usgovcloudapi.net` in Azure USGovernment

### Use case: 
Telegraf doesn't send metrics to the correct URL in non-Commercial Azure environments.  Allowing for an override value that, if set, changes the output URL for metrics data POSTs, would solve this roadblock for me (_and possibly other users who wish to leverage Azure Monitor in sovereign Azure cloud environments_) 

Microsoft does offer a method for determining the cloud environment you're operating in, via this method:
https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service#getting-azure-environment-where-the-vm-is-running

**Comment by ernstae** [comment]
I took a quick stab at the code necessary (minus any unit tests) in this attempt:

https://github.com/influxdata/telegraf/compare/master...ernstae:fix/issue-5452

**Comment by danielnelson** [comment]
We have something very similar in the `cloudwatch` output:
```
  ## Endpoint to make request against, the correct endpoint is automatically
  ## determined and this option should only be set if you wish to override the
  ## default.
  ##   ex: endpoint_url = "http://localhost:8000"
  # endpoint_url = ""
```

**Comment by ernstae** [comment]
I just updated the PR to switch the variable naming to `endpoint_url` for consistency-sake @danielnelson

**Comment by danielnelson** [comment]
Closed in #5453
